Documentation/Contributing.md

```diff
@@ -0,0 +1,348 @@
+
+## Reporting issues
+Report a bug guide: https://github.com/apple/swift-package-manager/blob/main/Documentation/Resources.md#reporting-a-good-swiftpm-bug.  
+JIRA Bug Tracker (a place where you can open bugs, enhancements to start to contribute): [https://bugs.swift.org/browse/SR-13640?jql=component%20%3D%20%22Package%20Manager%22](https://bugs.swift.org/browse/SR-13640?jql=component%20%3D%20%22Package%20Manager%22).
+
+## Development environment
+
+SwiftPM is typically built with a pre-existing versions of SwiftPM present on the system. Which can be obtained by installing the latest release of Xcode or a toolchain from swift.org. But there are have multiple ways to setup your development environment:
+
+### Using Xcode (Easiest)
+
+1. Install latest Xcode from https://developer.apple.com/xcode
+2. Open SwiftPM's `Package.swift` manifest with the latest release (including betas) of Xcode.
+
+### Using a Swift toolchain
+
+1. Download a toolchain from https://swift.org/download/
+
+2. Confirm the toolchain is installed correctly:
+
+*macOS*
+```bash
+$> export TOOLCHAINS=swift
+```
+
+Verify that we're able to find the swift compiler from the installed toolchain.
+```bash
+$> xcrun --find swift
+/Library/Developer/Toolchains/swift-latest.xctoolchain/usr/bin/swift
+$> swift package --version
+Swift Package Manager - Swift 5.3.0
+$> swift --version
+Apple Swift version 5.3
+```
+
+*Linux*
+```bash
+$> export PATH=/path/to/swift-toolchain/usr/bin:"${PATH}"
+```
+Verify that we're able to find the swift compiler from the installed toolchain.
+```bash
+$> which swift
+/path/to/swift-toolchain/usr/bin/swift
+$> swift package --version
+Swift Package Manager - Swift 5.3.0
+$> swift --version
+Apple Swift version 5.3
+```
+
+3. Alternatively use tools like swiftenv that help manage toolchains versions.
+
+### Using the Swift Compiler Build Script
+
+Follow these instructions to get the Swift sources and then execute the build-script using swiftpm preset:
+
+*macOS*
+```bash
+$> ./swift/utils/build-script --preset=buildbot_swiftpm_macos_platform,tools=RA,stdlib=RA
+```
+
+*Linux*
+```bash
+$> ./swift/utils/build-script --preset=buildbot_swiftpm_linux_platform,tools=RA,stdlib=RA
+```
+
+3. Clone [llbuild](https://github.com/apple/swift-llbuild) beside the package manager directory.
+
+```bash
+$> git clone https://github.com/apple/swift-llbuild llbuild
+$> ls
+swiftpm llbuild
+```
+
+Note: Make sure the directory for llbuild is called "llbuild" and not
+    "swift-llbuild".
+
+4. Clone [Yams](https://github.com/jpsim/yams) beside the package manager directory.
+
+```bash
+$> git clone https://github.com/jpsim/yams
+```
+
+5. Clone [swift-driver](https://github.com/apple/swift-driver) beside the package manager directory.
+
+```bash
+$> git clone https://github.com/apple/swift-driver
+```
+
+6. Clone [swift-argument-parser](https://github.com/apple/swift-argument-parser) beside the package manager directory and check out tag 0.3.0.
```

Comment:
the version number should be the one specified in Package.swift so maybe instead of hard coding it here, say so and add link to the file

Reply:
Does this not referring to the tag version?

https://github.com/apple/swift-argument-parser/tags